headers/crypto/1.2/psa/crypto-wpa3-sae.h

```diff
@@ -0,0 +1,313 @@
+#define PSA_KEY_TYPE_WPA3_SAE_GROUP_MASK       ((psa_key_type_t) 0x00ff)
+
+/** WPA3-SAE-PT key.
+ *
+ * The key is used to store the out of band calculated group element
+ * used in the Hash-To-Element variant of WPA3-SAE. It can be used as
+ * input to the WPA3-SAE PAKE instead of a password key.
+ *
+ * \param group A value of type ::psa_ec_family_t or ::psa_dh_family_t
+ *              that identifies the group to be used.
+ */
+#define PSA_KEY_TYPE_WPA3_SAE_PT(group) \
+    ((psa_key_type_t) (PSA_KEY_TYPE_WPA3_SAE_PT_BASE | (group)))
+
+ /** Whether a key type is a WPA3-SAE-PT. */
+#define PSA_KEY_TYPE_IS_WPA3_SAE_PT(type)                    \
+    (((type) & ~PSA_KEY_TYPE_WPA3_SAE_GROUP_MASK) ==         \
+     PSA_KEY_TYPE_WPA3_SAE_PT_BASE)
+ /** Extract the group from a WPA3-SAE key type. */
+#define PSA_KEY_TYPE_WPA3_SAE_PT_GET_FAMILY(type)            \
+    ((psa_ecc_family_t) (PSA_KEY_TYPE_IS_WPA3_SAE_PT(type) ? \
+      ((type) & PSA_KEY_TYPE_WPA3_SAE_GROUP_MASK) :          \
+      0))
+
+#define PSA_ALG_WPA3_SAE_PT_BASE          ((psa_algorithm_t) 0x08800400)
+/** The WPA3-SAE password to PT KDF.
+ * It takes the password p, a salt (uuid), and optionally a password id.
+ * 
+ * This key derivation algorithm uses the following inputs, which must be
+ * provided in the following order:
+ * - #PSA_KEY_DERIVATION_INPUT_SALT for the uuid.
+ * - #PSA_KEY_DERIVATION_INPUT_SECRET for the password.
+ * - optionally; #PSA_KEY_DERIVATION_INPUT_INFO for the password id.
+ * The output has to be read as a key of type PSA_KEY_TYPE_WPA3_SAE_PT.
+ *
+ * \param hash_alg      A hash algorithm (\c PSA_ALG_XXX value such that
+ *                      #PSA_ALG_IS_HASH(\p hash_alg) is true).
+ *
+ * \return              The corresponding counter-mode KDF algorithm.
+ * \return              Unspecified if \p hash_alg is not a supported
+ *                      hash algorithm.
+ */
+#define PSA_ALG_WPA3_SAE_PT(hash_alg)                            \
+    (PSA_ALG_WPA3_SAE_PT_BASE | ((hash_alg) & PSA_ALG_HASH_MASK))
+
+/** Whether the specified algorithm is a key derivation algorithm constructed
+ * using #PSA_ALG_WPA3_SAE_PT(\p hash_alg).
+ *
+ * \param alg An algorithm identifier (value of type #psa_algorithm_t).
+ *
+ * \return 1 if \p alg is a key derivation algorithm constructed using #PSA_ALG_WPA3_SAE_PT(),
+ *         0 otherwise. This macro may return either 0 or 1 if \c alg is not a supported
+ *         key derivation algorithm identifier.
+ */
+#define PSA_ALG_IS_WPA3_SAE_PT(alg)                         \
+    (((alg) & ~PSA_ALG_HASH_MASK) == PSA_ALG_WPA3_SAE_PT_BASE)
+
+/** The WPA3-SAE key exchange algorithm.
+ *
+ * This is WPA3-SAE as defined by "IEEE Std 802.11-REVme/D7.0 2024, 
+ * chapter 12.4", including the Hash-To-Element (H2E) variant.
+ * It is instantiated with the following parameters:
+ *
+ * - The group is defined over a finite field or an elliptic curve.
+ * - A cryptographic hash function.

```

Comment:
See comment against L44 regarding the need to explicitly specify the hash parameter?

Reply:
see above